### PR TITLE
Fix SDPA bool mask producing NaN for fully masked rows

### LINKED ISSUE #35959: [Bug]: OpenVINO converts PyTorch scaled_dot_product_attention bool mask to NaN output
### OpenVINO Version

2026.1.0

### Operating System

Other (Please specify in description)

### Device used for inference

CPU

### Framework

ONNX

### Model used

Minimal PyTorch model converted by ov.convert_model.

### Issue description

OpenVINO CPU returns `NaN` for a PyTorch model containing `torch.nn.functional.scaled_dot_product_attention` with an all-False boolean attention mask.

For the same input, PyTorch returns a finite all-zero tensor, while the OpenVINO-converted model returns all `NaN`.

OS: Ubuntu 24.04.2 LTS / Linux x86_64

### Step-by-step reproduction

```
import sys
import numpy as np
import openvino as ov
import torch

class M(torch.nn.Module):
    def forward(self, q, k, v, mask):
        return torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask)

torch.manual_seed(13)

model = M().eval()

q = torch.randn(1, 1, 3, 4)
k = torch.randn(1, 1, 3, 4)
v = torch.randn(1, 1, 3, 4)
mask = torch.zeros(1, 1, 3, 3, dtype=torch.bool)

torch_out = model(q, k, v, mask).detach().numpy()

ov_model = ov.convert_model(
    model,
    example_input={
        "q": q,
        "k": k,
        "v": v,
        "mask": mask,
    },
)

compiled = ov.Core().compile_model(ov_model, "CPU")

openvino_out = compiled(
    {
        "q": q.numpy(),
        "k": k.numpy(),
        "v": v.numpy(),
        "mask": mask.numpy(),
    }
)[compiled.outputs[0]]

print("q_shape:", tuple(q.shape))
print("mask_all_false:", bool((~mask).all()))
print("torch:", torch_out)
print("openvino_cpu:", openvino_out)
print("torch_finite:", bool(np.isfinite(torch_out).all()))
print("openvino_finite:", bool(np.isfinite(openvino_out).all()))

if np.isfinite(torch_out).all() and not np.isfinite(openvino_out).all():
    print("\nBUG REPRODUCED: OpenVINO returns NaN for converted SDPA bool-mask model.")
    sys.exit(0)

print("\nNOT REPRODUCED")
sys.exit(1)
```

### Relevant log output

```shell
Expected:
OpenVINO CPU should return a finite output consistent with PyTorch for scaled_dot_product_attention with an all-False boolean attention mask.

Actual:
q_shape: (1, 1, 3, 4)
mask_all_false: True
torch: [[[[0. 0. 0. 0.]
   [0. 0. 0. 0.]
   [0. 0. 0. 0.]]]]
openvino_cpu: [[[[nan nan nan nan]
   [nan nan nan nan]
   [nan nan nan nan]]]]
torch_finite: True
openvino_finite: False

BUG REPRODUCED: OpenVINO returns NaN for converted SDPA bool-mask model.
```

### Issue submission checklist

- [x] I'm reporting an issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

### PR DESCRIPTION
**Pull request by ssam18**
### Details
- Add a fully masked row guard after softmax in the ScaledDotProductAttention decomposition so rows where the boolean attn_mask is entirely False return zeros instead of NaN, matching PyTorch eager.
- Mirror the guard in the reference implementation and in the intel_cpu softmax kernel where it short circuits to a zero row when max equals minus infinity, applied across the float and float16 specializations.
- Add a pytest case in tests/layer_tests/pytorch_tests/test_scaled_dot_product_attention.py exercising the all False bool mask scenario that the reporter shared, since current parametrized tests only use random 0/1 masks and never hit the fully masked row path.

### Tickets
- Fixes #35959

### CODE REVIEW

**Comment by maxnick** [comment]
@zhangYiIntel , could you please review?

File: src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp [hunk 1]
```diff
@@ -185,6 +187,9 @@ std::shared_ptr<ov::Node> ov::pass::ScaledDotProductAttentionDecomposition::deco
         scaled_atten = register_new_node<v8::Softmax>(scaled_atten, -1);
     }
 
+    auto is_nan = register_new_node<v10::IsNaN>(scaled_atten);
+    scaled_atten = register_new_node<v1::Select>(is_nan, zero_f, scaled_atten)->output(0);
+
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/op_conversions/scaled_dot_product_attention_decomposition.cpp]: Do we need transformations test here?

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 2]
```diff
@@ -1231,6 +1231,17 @@ inline void attn_softmax_kernel<float>(float* a,
     if (sink != nullptr) {
         max = max > (*sink) ? max : (*sink);
     }
+
```
Comment (zhangYiIntel) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: Hello, one question about the `mask` to sdpa, during the inference is it converted to float ? 
In normal [SDPA kernel](https://github.com/openvinotoolkit/openvino/blob/6a1298fecdd1bf41fb33306aa4d428ac60bf81f1/src/plugins/intel_cpu/src/nodes/scaled_attn.cpp#L1470),  it converts boolean mask to float here. But I am not sure in your test whether the `mask` is already NaN before fed into SDPA node.
```
    void prepare_attn_mask(const MemoryPtr& attn_input) {
        attn_buf.resize<float>(attn_input->getStaticDims());
        auto* p = attn_input->getDataAs<uint8_t>();
        for (size_t i = 0; i < attn_input->getSize(); i++) {
            attn_buf.ptr<float>()[i] = p[i] ? 0.0F : -FLT_MAX;
        }
    }
```

File: src/core/reference/include/openvino/reference/scaled_dot_product_attention.hpp
Comment (praasz) [inline src/core/reference/include/openvino/reference/scaled_dot_product_attention.hpp:1]: The tests in `src/plugins/template/tests/functional/op_reference/scaled_dot_product_attention.cpp`
should be updated after changes in reference

**Review by zhangYiIntel — CHANGES_REQUESTED** [review]
@ssam18 I want to point out 1 thing

- `attention_mask` with bool type is supported by decomposition `ScaledDotProductAttentionDecomposition`, therefore the fix should be at `scaled_dot_product_attention_decomposition.cpp`,  the code in plugin shouldn't be applied.

CC: @maxnick @isanghao

File: src/core/reference/include/openvino/reference/scaled_dot_product_attention.hpp
Comment (ssam18) [reply src/core/reference/include/openvino/reference/scaled_dot_product_attention.hpp:1]: Done in 11731e8a — added smoke_SDPA_With_Fully_Masked_Row in scaled_dot_product_attention.cpp covering the new fully-masked-row path in the reference, parameterized on f32/f16/f64.

**Comment by ssam18** [comment]
@zhangYiIntel Thanks for flagging. The bool-mask fix is in `scaled_dot_product_attention_decomposition.cpp` (the `has_bool_mask` / `fully_masked_rows` logic) and the matching tests are in `scaled_dot_product_decomposition_test.cpp`, exactly where you would expect. The `softmax_kernel.hpp` change was a stray trailing-newline added by my editor on file open and had no functional effect.
